### ISSUE TITLE
[MR] 코드리뷰 - [merged]

### ISSUE THREAD

**Issue by seonghbae** [seonghbae/TogeFit#291]
In GitLab by @ryeong904 on Jul 15, 2022, 22:57

_Merges dev -> master_

**Comment by seonghbae** [comment]
In GitLab by @jinwoo on Jul 15, 2022, 23:13

added 2 commits

<ul><li>fb3f6154 - [#111] feat: 권한 401 핸들링</li><li>edc7e320 - Merge branch &#39;feature/#111&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37484&start_sha=603a4446a60a9197a79c11a3088654707fe71a24)

**Comment by seonghbae** [comment]
In GitLab by @jinwoo on Jul 15, 2022, 23:32

added 3 commits

<ul><li>0c792477 - [#114] feat: 게시글 인터페이스</li><li>016ef0d2 - [#114] feat: 게시글 인터페이스 export</li><li>2c0e4eb5 - Merge branch &#39;feature/#114&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37491&start_sha=edc7e3200026dddfe580f4425a4bb82789465098)

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:24

Commented on [client/src/common/components/alert-modal/index.tsx line 8](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-4725a6130ab5d5fdd9b77abaad98b375R8)



```
handleConfirm(): void;
handleCancel?(): void;
```
이거도 가능

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:27

Commented on [client/src/common/components/alert-modal/index.tsx line 40](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-4725a6130ab5d5fdd9b77abaad98b375R40)



엘리먼트에 이와같이 이벤트 핸들러를 작성하셨는데,
함수로 따로 빼시는게 좋음 

현재는 단순한 코드라 읽어나가는게 가능하나, 이부분이 커지면 JSX 구문 오염 및 가독성이 떨어짐

```
onClick={e => xxxx(e)}

const xxxx = e => {
  if (e.target === ref.current) {
          if (handleCancel) handleCancel();
          else handleConfirm();
  }
};

```

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:30

Commented on [client/src/common/components/custom-card/CustomCard.tsx line 21](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-215094f6e4d73920b538c512ca001fe4R21)



nanoId
uuid 와 같은 모듈을 사용하시는게 좋음

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:34

Commented on [client/src/common/utils/getPath.ts line 5](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-3674c4cc816e0e6955ed516bf549e764R5)



at() 쓴거 칭찬해!

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:37

Commented on [client/src/pages/add-routine-page/components/AddExerciseModalStyle.tsx line 15](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-84d34950debd35e7db89870d65769ca5R15)



모달같은거는 z-index 9999 잡는걸 추천

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:43

코드를 살펴보면, 전체적으로 Javascript 최신 스펙에 대한 이해도가 있다고 생각이듬

axios 처리 부분을
async await 구문으로 처리하면 보다 간결하고 읽기 좋은 코드가 될거같음(Promise finally 도 잘 사용함)
```
async () => {
  try {
    const result1 = await firstAsynchronousFunction();
    const result2 = await secondAsynchronousFunction(result1);
    console.log(result2);
  } catch(err) {
    throw new Error(`Something failed`);
  } finally {
    console.log(`All Tasks are Done`);
  }
}
```

커스텀훅을 사용하여 api 로직을 잘 분리한거는 잘하신거임
근데 매번 API 요청이 늘어날때마다 커스텀훅이 늘어나고, isLoading 이라던가 보일러플레이트 코드가 발생을 할거같음
지금도 잘하셨지만, 이 부분을 어떻게 공통화를 시킬까 고민을 해보면 좋음

**Comment by seonghbae** [comment]
In GitLab by @chbae on Jul 16, 2022, 11:43

approved this merge request

**Comment by seonghbae** [comment]
In GitLab by @coy on Jul 16, 2022, 12:52

Commented on [server/src/routers/PostRouter.ts line 103](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-5f2343f628a7835594403be8f5d55022R103)



둘중에 하나라도 없으면 디폴트로 오늘 날을 넣어주는 로직같은데 둘중에 하나만 없고 하나는 잘들어왔을땐 어떻게 하나요?

**Comment by seonghbae** [comment]
In GitLab by @coy on Jul 16, 2022, 12:52

Commented on [server/src/services/UserService.ts line 159](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..2c0e4eb55114c24f4d150698f36b0fcbb6ffa056#diff-4a22707e4fd843e63260d25c3bcb7fd3R159)



오피스아워때 이야기한대로 400이 좋아보입니다

**Comment by seonghbae** [comment]
In GitLab by @coy on Jul 16, 2022, 12:52

Commented on [server/tests/food.test.ts line 1](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-7f01144e62d87df74b479d479d973e65R1)



각각해당하는 실패시나리오가 없는 describe들은 다 채워넣어주세요

**Comment by seonghbae** [comment]
In GitLab by @Circlewee on Jul 16, 2022, 15:35

Commented on [client/src/common/components/alert-modal/index.tsx line 40](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-4725a6130ab5d5fdd9b77abaad98b375R40)



테스트해본다고 그냥 안에 넣어둔걸 그대로 올렸었네요 분리하겠습니다

**Comment by seonghbae** [comment]
In GitLab by @Circlewee on Jul 16, 2022, 15:35

Commented on [client/src/common/components/custom-card/CustomCard.tsx line 21](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-215094f6e4d73920b538c512ca001fe4R21)



넵 이부분도 수정하겠습니다

**Comment by seonghbae** [comment]
In GitLab by @Circlewee on Jul 16, 2022, 15:35

Commented on [client/src/pages/add-routine-page/components/AddExerciseModalStyle.tsx line 15](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-84d34950debd35e7db89870d65769ca5R15)



어디서든 맨위에 위치해야하니 혹시모를 상황을 대비해 낫겠군요

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 16, 2022, 16:29

Commented on [server/tests/food.test.ts line 1](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-7f01144e62d87df74b479d479d973e65R1)



실패 시나리오를 추가하라는 말씀이신건가요? 일단 router 단에서 처리되는 케이스가 많아서 service 단에서 실패하는 시나리오를 아직 생각해보지 못했습니다.. 일단 더 생각해보겠습니다!

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 16, 2022, 16:29

Commented on [server/src/services/UserService.ts line 159](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..2c0e4eb55114c24f4d150698f36b0fcbb6ffa056#diff-4a22707e4fd843e63260d25c3bcb7fd3R159)



네 수정하겠습니다.

**Comment by seonghbae** [comment]
In GitLab by @ryeong904 on Jul 16, 2022, 16:50

Commented on [server/src/routers/PostRouter.ts line 103](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..fc693777c52a091e94508d881acc335341885fd3#diff-5f2343f628a7835594403be8f5d55022R103)



현재는 하나만 잘 들어왔을 때도 오늘 날을 넣어줍니다.

하나는 없고 하나만 있는 경우 잘 들어온 값을 기준으로 값을 돌려주도록 수정이 필요할 것 같습니다.

year=2022만 들어오면 오늘의 월(month=7)로 검색하고, month=5만 들어오면 오늘의 연도(year=2022)로 검색하는 형식으로 수정하겠습니다.

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 16, 2022, 16:51

Commented on [server/src/services/UserService.ts line 159](https://github.com/seonghbae/TogeFit/compare/9881d4709c870c12b61712b25d707aaa6a611de9..2c0e4eb55114c24f4d150698f36b0fcbb6ffa056#diff-4a22707e4fd843e63260d25c3bcb7fd3R159)



changed this line in [version 4 of the diff](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37762&start_sha=2c0e4eb55114c24f4d150698f36b0fcbb6ffa056#64cf4e5ba30d64a9fefcbdc977d63749673b8b0f_159_155)

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 16, 2022, 16:51

added 5 commits

<ul><li>17cac93f - [#102]refactor: 사용하지 않는 함수 제거</li><li>9090a0fc - [#105]fix: 상태코드 수정</li><li>deec92d9 - [#105]feat: 게시글 테스트 코드 작성</li><li>7e9a4f07 - [#105]feat: 게시글 테스트 코드 - 실패 케이스 추가</li><li>ff70daa6 - Merge branch &#39;feature/#105&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37762&start_sha=2c0e4eb55114c24f4d150698f36b0fcbb6ffa056)

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 16, 2022, 18:50

added 3 commits

<ul><li>fe777c37 - [#102]fix: 예외사항 추가</li><li>8a2653ac - [#102]feat: 운동리스트 테스트 코드 작성</li><li>0c9c6c98 - Merge branch &#39;feature/#102&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37800&start_sha=ff70daa6f1a7402859df38c3b301c4185c5b0310)

**Comment by seonghbae** [comment]
In GitLab by @ryeong904 on Jul 17, 2022, 24:14

added 3 commits

<ul><li>bd4eeeda - [#116]feat: Meal 테스트 코드 작성</li><li>239b6095 - [#116]fix: 디버깅용 console.log 삭제</li><li>67dd6e14 - Merge branch &#39;feature/#116&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=37907&start_sha=0c9c6c989432c53121018afe4547aea419816578)

**Comment by seonghbae** [comment]
In GitLab by @jinwoo on Jul 18, 2022, 10:27

added 5 commits

<ul><li>a4e13139 - [#112] feat: 메인화면 구성 및 기본 api 호출</li><li>84bf13f2 - [#112] style: 코드정리</li><li>2fd5b919 - [#112] fix: 잘못된 커밋 rollback</li><li>efbca3ec - [#112] style: 수정</li><li>fc693777 - Merge branch &#39;feature/#112&#39; into &#39;dev&#39;</li></ul>

[Compare with previous version](/sw_track/class_02_seoul/web_project_2/team16/healthcare-community/-/merge_requests/98/diffs?diff_id=38296&start_sha=67dd6e1473d569bac6e2bd54bf098b8ded0c0d4d)

**Comment by seonghbae** [comment]
In GitLab by @KimMiso on Jul 18, 2022, 11:09

mentioned in commit 985fe011f60e532f1b9448edbdac77e40411c050
